pipelines/services/sync.go

```diff
@@ -0,0 +1,117 @@
+package services
+
+import (
+	"encoding/json"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+)
+
+type SyncService struct {
+	client *jfroghttpclient.JfrogHttpClient
+	auth.ServiceDetails
+}
+
+func (ss *SyncService) getHttpDetails() httputils.HttpClientDetails {
+	httpDetails := ss.ServiceDetails.CreateHttpClientDetails()
+	return httpDetails
+}
+
+func NewSyncService(client *jfroghttpclient.JfrogHttpClient) *SyncService {
+	return &SyncService{client: client}
+}
+
+// SyncPipelineSource trigger sync for pipeline resource
+func (ss *SyncService) SyncPipelineSource(branch string, repoName string) (int, []byte, error) {
+	// fetch resource ID
+	resID, _, resourceErr := ss.GetPipelineResourceID(repoName)
+	if resourceErr != nil {
+		log.Error("unable to fetch resourceID for: ", repoName)
+		return 0, []byte{}, errorutils.CheckError(resourceErr)
+	}
+	log.Info("triggering pipeline source sync ...")
+
+	// trigger sync
+	httpDetails := ss.getHttpDetails()
+	queryParams := make(map[string]string, 0)
+	queryParams["sync"] = "true"
+	queryParams["branch"] = branch
+
+	apiPath := path.Join(pipelineResources, strconv.Itoa(resID))
+	uriVal, errURL := ss.constructURL(apiPath, queryParams)
+	if errURL != nil {
+		return 0, []byte{}, errorutils.CheckError(errURL)
+	}
+	resp, body, _, httpErr := ss.client.SendGet(uriVal, true, &httpDetails)
+	if httpErr != nil {
+		return 0, body, errorutils.CheckError(httpErr)
+	}
+	if err := errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK); err != nil {
+		return 0, body, errorutils.CheckError(err)
+	}
+	if resp.StatusCode == http.StatusOK {
+		log.Info("Triggered pipeline sync successfully")
+	}
+	return resp.StatusCode, body, nil
+}
+
+// GetPipelineResourceID fetches resource ID for given full repository name
+func (ss *SyncService) GetPipelineResourceID(repoName string) (int, bool, error) {
+	httpDetails := ss.getHttpDetails()
+	queryParams := make(map[string]string, 0)
+
+	uriVal, errURL := ss.constructURL(pipelineResources, queryParams)
+	if errURL != nil {
+		return 0, false, errorutils.CheckError(errURL)
+	}
+
+	resp, body, _, err := ss.client.SendGet(uriVal, true, &httpDetails)
+	if err != nil {
+		return 0, false, errorutils.CheckError(err)
+	}
+	// Response Analysis
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK); err != nil {
+		return 0, false, err
+	}
+	if resp.StatusCode == http.StatusOK {
+		log.Debug("received resource id")
+	}
```

Comment:
Tautology + capitalize
```suggestion
        log.Debug("Received resource id")
```